lib/env_checker/configuration.rb

```diff
@@ -6,23 +6,39 @@ module EnvChecker
   class ConfigurationError < StandardError; end
 
   class Configuration
-    attr_accessor :required_variables, :optional_variables, :logger,
-                  :environment, :slack_webhook_url, :slack_notifier,
-                  :environments
+    ATTRIBUTES = %w(required_variables optional_variables environment
+                    config_file slack_webhook_url environments).freeze
+    OBJECTS = %w(logger slack_notifier configurations).freeze
```

Comment:
Using `%i` instead of `%w` would avoid having to later call `.map(&.to_sym)`, for example:

```ruby
    ATTRIBUTES = %i(required_variables optional_variables environment
                    config_file slack_webhook_url environments).freeze
```